llvm/utils/llvm_push_pr.py

```diff
@@ -0,0 +1,653 @@
+            text=True,
+            read_only=True,
+        )
+        commits = result.stdout.strip().split("\n")
+        return [c for c in commits if c]
+
+    def _get_commit_details(self, commit_hash: str) -> tuple[str, str]:
+        """Gets the title and body of a commit."""
+        result = self._run_cmd(
+            ["git", "show", "-s", "--format=%s%n%n%b", commit_hash],
+            capture_output=True,
+            text=True,
+            read_only=True,
+        )
+        parts = result.stdout.strip().split("\n\n", 1)
+        title = parts[0]
+        body = parts[1] if len(parts) > 1 else ""
+        return title, body
+
+    def _sanitize_for_branch_name(self, text: str) -> str:
+        """Sanitizes a string to be used as a git branch name."""
+        sanitized = re.sub(r"[^\w\s-]", "", text).strip().lower()
+        sanitized = re.sub(r"[-\s]+", "-", sanitized)
+        # Use "auto-pr" as a fallback.
+        return sanitized or "auto-pr"
+
+    def _create_and_push_branch_for_commit(
+        self, commit_hash: str, base_branch_name: str, index: int
+    ) -> str:
+        """Creates and pushes a temporary branch pointing to a specific commit."""
+        branch_name = f"{self.args.prefix}{base_branch_name}-{index + 1}"
+        commit_title, _ = self._get_commit_details(commit_hash)
+        self.printer.print(f"Processing commit {commit_hash[:7]}: {commit_title}")
+        self.printer.print(f"Creating and pushing temporary branch '{branch_name}'")
+
+        self._run_cmd(["git", "branch", "-f", branch_name, commit_hash])
+        push_command = ["git", "push", self.args.remote, branch_name]
+        self._run_cmd(push_command)
+        self.created_branches.append(branch_name)
+        return branch_name
+
+    def run(self):
+        """Main entry point for the automator, orchestrates the PR creation and merging process."""
+        self.repo_slug = self._get_repo_slug()
+        self.repo_settings = self.github_api.get_repo_settings()
+        self.original_branch = self._get_current_branch()
+        self.printer.print(f"On branch: {self.original_branch}")
+
+        try:
+            self._rebase_current_branch()
+            initial_commits = self._get_commit_stack()
+
+            if not initial_commits:
+                self.printer.print("No new commits to process.")
+                return
+
+            if self.args.auto_merge and len(initial_commits) > 1:
+                self.printer.print(
+                    "Error: --auto-merge is only supported for a single commit.",
+                    file=sys.stderr,
+                )
+                sys.exit(1)
+
+            if self.args.no_merge and len(initial_commits) > 1:
+                self.printer.print(
+                    "Error: --no-merge is only supported for a single commit. "
+                    "For stacks, the script must merge sequentially.",
+                    file=sys.stderr,
+                )
+                sys.exit(1)
```

Comment:
Why is that? When pushing branches to upstream, you can cascade pull-requests to create your stack.